packages/admin/blocks-admin/src/blocks/types.tsx

```diff
@@ -85,7 +85,7 @@ export interface BlockMethods<
     dependencies?: (state: State) => BlockDependency[];
     replaceDependenciesInOutput: (output: OutputApi, replacements: ReplaceDependencyObject[]) => OutputApi;
     resolveDependencyPath: (state: State, jsonPath: string) => string;
-    extractTextContents?: (state: State) => string[];
+    extractTextContents?: (state: State, options?: { includeInvisibleContent: boolean }) => string[];
```

Comment:
Instead of having an options prop, the could change the return type to `{ content: string; visible: boolean; }`. This would force devs to think about the visibility of the content and prevent that they forget considering the option. What do you think?

Reply:
> This would force devs to think about the visibility of the content and prevent that they forget considering the option.

👍 generally a good idea. But it's a bit harder to implement since the visibility of a block depends on the visibility of its parent. So I must either A) find a way to provide the visibility of the parent to the children or B) override the visible prop for the children.

B is probably easier to do, so I'll implement this as an example

Reply:
I prefer B as well 🙂

Reply:
You can't really handle indirect visibility with that.

Reply:
I would stick with the current implementation, I don't see any problem with it

Reply:
>  prevent that they forget considering the option

So you think this could be neglected?

Reply:
> So you think this could be neglected?

imho, yes.

- most apps won't need their own implementation that needs to check visibility
- you can always forget something (with the visible returned, you can forget about indirect visibility)

Reply:
Make options required

Reply:
fa5c13cb9ac8de188f8c960b9849fb2c3a209a58